x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/InferenceSettingsTestCase.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.services;
+
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.test.AbstractWireSerializingTestCase;
+import org.elasticsearch.xcontent.ToXContent;
+import org.elasticsearch.xcontent.ToXContentFragment;
+import org.elasticsearch.xcontent.XContentParserConfiguration;
+import org.elasticsearch.xcontent.json.JsonXContent;
+import org.elasticsearch.xpack.inference.InferenceNamedWriteablesProvider;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.equalTo;
+
+/**
+ * All Inference Setting classes derived from {@link org.elasticsearch.inference.ServiceSettings},
+ * {@link org.elasticsearch.inference.TaskSettings}, and {@link org.elasticsearch.inference.SecretSettings}
+ * must be able to read/write to an index via ToXContent as well as read/write between nodes via Writeable.
+ */
+public abstract class InferenceSettingsTestCase<T extends Writeable & ToXContent> extends AbstractWireSerializingTestCase<T> {
```

Comment:
How about we use `AbstractBWCWireSerializationTestCase`, I think it might cover more 🤔